### ISSUE TITLE
support for 64 bits integer using bit64 and integer64

### ISSUE THREAD

**Issue by yimyom** [tidyverse/readr#633]
Hi,

would it be possible to have a support for 64 bits integer right out of the box. It is very common nowadays to have nanoseconds time stamps precision and therefore, they are encoded as 64 bits.
For example, these are the kind of values I use all the time: 1487811688746493361 and they are read by default as doubles.

Example of code:
```
> read_csv("a\n1487810827837636484",col_types="i")
Warning: 1 parsing failure.
row col   expected              actual
  1   a an integer 1487810827837636484

# A tibble: 1 × 1
      a
  <int>
1    NA
> read_csv("a\n1487810827837636484",col_types="n")                                                                
# A tibble: 1 × 1
             a
         <dbl>
1 1.487811e+18
> read_csv("a\n1487810827837636484",col_types="d")                                                                
# A tibble: 1 × 1
             a
         <dbl>
1 1.487811e+18
```

The first example fails. If a `col_types` for 64 bits integer (like `l`), it would work and would return a vector of integer64 (from the bit64 package). The other example suffers a loss of accuracy.

For the time being, I am forced to read the column as characters and convert them after to integer64.
```
> x<-read_csv("a\n1487810827837636484",col_types="c")
> x
# A tibble: 1 × 1
                    a
                <chr>
1 1487810827837636484
> x$a=as.integer64(x$a)
> x
# A tibble: 1 × 1
                    a
      <S3: integer64>
1 1487810827837636484
```

**Comment by WatanabeSmith** [comment]
This might be tangential, but given the age of this thread I'm throwing out a workaround I encountered in the hopes that it helps.

I have a large .rds I'm using (created from database queries with `RMariaDB`). When read it with `read_rds` it mangles the integer64 fields, but if I `library(RMariaDB)` then it works fine. I assume `RMariaDB` is loading some int64 support that carries over into `readr`.

[https://imgur.com/a/qAbmxXK](screencap) - Relevant screen capture, I can't manage to make a reprex, not sure how to remake this .rds on the fly. But also I'm not looking for assistance, I have my workaround, just hoping that this helps.

**Comment by jimhester** [comment]
Fixed by #1172
